test/luatest_helpers/cluster.lua

```diff
@@ -72,9 +72,22 @@ function Cluster:stop()
 end
 
 function Cluster:start(opts)
-    for _, server in ipairs(self.servers) do
+    local replication_settings = {}
+
+    for i, server in ipairs(self.servers) do
         if not server.process then
-            server:start({wait_for_readiness = false})
+            if server.box_cfg ~= nil and server.box_cfg.replication ~= nil then
+                replication_settings[i] = server.box_cfg.replication
+                server.box_cfg.replication = self.servers[1].net_box_uri
+            end
+            server:start({wait_for_readiness = true})
+        end
+    end
+
+    for i, server in ipairs(self.servers) do
+        server:box_config({replication = replication_settings[i]})
```

Comment:
I guess this should go under the `if replication_settings[i] ~= nil`

Reply:
Thanks for review! There was a bigger problem with that part of the code. I wanted that after start() has finished server.box_cfg was the same as before (as it was set by user) and box.cfg on the server itself was in the state user asked it to be. I've rewritten it, take a look please.